packages/hash/api/src/model/knowledge/link.model.ts

```diff
@@ -0,0 +1,160 @@
+import { GraphApi } from "@hashintel/hash-graph-client";
+
+import { EntityModel, LinkModel, LinkTypeModel } from "../index";
+
+export type LinkModelConstructorParams = {
+  accountId: string;
+  sourceEntity: EntityModel;
+  linkTypeModel: LinkTypeModel;
+  targetEntity: EntityModel;
+};
+
+export type LinkModelCreateParams = {
+  accountId: string;
+  sourceEntity: EntityModel;
+  linkTypeModel: LinkTypeModel;
+  targetEntity: EntityModel;
+};
+
+/**
+ * @class {@link LinkModel}
+ */
+export default class {
+  accountId: string;
+
+  sourceEntity: EntityModel;
+  linkTypeModel: LinkTypeModel;
+  targetEntity: EntityModel;
+
+  constructor({
+    accountId,
+    sourceEntity,
+    linkTypeModel,
+    targetEntity,
+  }: LinkModelConstructorParams) {
+    this.accountId = accountId;
+
+    this.sourceEntity = sourceEntity;
+    this.linkTypeModel = linkTypeModel;
+    this.targetEntity = targetEntity;
+  }
+
+  /**
+   * Create a link between a source and a target entity using a specific link
+   * type.
+   *
+   * @param params.accountId the accountId of the account creating the link
+   * @param params.sourceEntity the source entity of the link
+   * @param params.linkTypeModel the Link Type of the link
+   * @param params.targetEntity the target entity of the link
+   */
+  static async create(
+    graphApi: GraphApi,
+    {
+      accountId,
+      sourceEntity,
+      linkTypeModel,
+      targetEntity,
+    }: LinkModelCreateParams,
+  ): Promise<LinkModel> {
+    const {
+      data: { sourceEntityId, linkTypeUri, targetEntityId },
+    } = await graphApi.createLink(sourceEntity.entityId, {
+      accountId,
+      linkTypeUri: linkTypeModel.schema.$id,
+      targetEntityId: targetEntity.entityId,
+    });
+
+    const fetchedParams = {
+      sourceEntity: await EntityModel.getLatest(graphApi, {
+        accountId,
+        entityId: sourceEntityId,
+      }),
+      linkTypeModel: await LinkTypeModel.get(graphApi, {
+        versionedUri: linkTypeUri,
+      }),
+      targetEntity: await EntityModel.getLatest(graphApi, {
+        accountId,
+        entityId: targetEntityId,
+      }),
+    };
+
+    return new LinkModel({
+      accountId,
+      ...fetchedParams,
+    });
+  }
+
+  /**
+   * Get all outgoing links of a source entity.
+   *
+   * @param params.accountId the accountId of the account creating the link
+   * @param params.sourceEntity the source entity of the link
+   */
+  static async getAllOutgoing(
+    graphApi: GraphApi,
+    { sourceEntity }: { sourceEntity: EntityModel },
+  ): Promise<LinkModel[]> {
+    const { data: entityLinks } = await graphApi.getEntityLinks(
+      sourceEntity.entityId,
+    );
+    return Promise.all(
+      Object.entries(entityLinks.outgoing)
+        /**
+         * When target entity id is a list the link has many targets.
+         * This is not supported yet
+         * @todo support 1:N links
```

Comment:
I don't understand this, why do we keep referring to this as a single link to multiple things? Is that how it was modelled prior to this rework?

I see this as just multiple links, where they all happen to share the same source. 

I don't see _**any**_ value of thinking about _a_ link having multiple targets. It seems like it would greatly confuse versioning, representations, and indeed this model class wouldn't make sense because it has `targetEntityModel` which is a singular thing.

Reply:
I refer to it as 1:N where one source entity links to multiple targets under the same link type. Not a single link to multiple things. I would like a better name for this, but I've settled with "1:N links" for now.

Reply:
Gotcha, I don't really mind us calling it `1:N` but I have more issues with phrasings like: "the link has many targets" because it's not a single link.

To be clear I'm not just being pedantic about phrasing here, I was concerned it was affecting the design of this model class. Should a Model class for _a specific_ **link** be concerned with getting multiple links? If that's the pattern we have for `EntityModel` and such then I don't mind because it's consistent, but I was a bit concerned that it was related to the aforementioned point.

Reply:
That's a valid concern. It is consistent with `EntityModel` fetching multiple entities (e.g. when calling `getAllLatest` `EntityModel[]` is returned, meaning multiple entities). 
I do think there will be some annoyances putting single and multi links in the same model class like we are doing here. Perhaps having a separate `MultiLinkModel` may make sense. As we don't need multi links right now, I don't think it's fair to make the assumption that it's useful to split them now, though.